packages/@sanity/components/src/panes/DefaultPane.js

```diff
@@ -295,41 +297,78 @@ class Pane extends React.Component {
     )
   }
 
+  renderActionMenuItem = item => {
+    const {styles} = this.props
+    if (!item) {
+      return null
+    }
+    const params = item.intent.params
+    const schemaType = S.documentTypeListItem(params.type).spec.schemaType
+    const icon = schemaType.icon
```

Comment:
This should preferably use the icon declared on the template (which falls back to the schema types icon, if not explicitly provided). 

I'm thinking it's perhaps best to do the resolving of this icon in the structure builders `S.menuItemsFromInitialValueTemplateItems()` method - replacing the `add` icon for the template icon, and falling back to the add icon?

```js
return new MenuItemBuilder()
  .title(title)
  .icon((tpl && tpl.icon) || getPlusIcon())
```

Reply:
OK, thank you! That makes a lot more sense. 

I did this, but now the icon of the second template is always undefined. How come? Shouldn't the fallback always be the icon of the schemaType the template is from, which is author? Or is there some logic somewhere that looks for an icon based on the `author-developer` template name?

<img width="226" alt="image" src="https://user-images.githubusercontent.com/25737281/66127741-aed8d800-e5ec-11e9-8e8c-e0539228de90.png">

This seems to give the right icon for both, but I'm guessing the problem is somewhere higher up in the chain and I can't find it.

```
const schema = getDefaultSchema().get(tpl.schemaType)
 return new MenuItemBuilder()
      .title(title)
      .icon((tpl && tpl.icon) || schema && schema.icon || getPlusIcon())
```

Reply:
Hmm, good question. I think we might have chosen not to fall back on the schema type icon so as not to repeat the same icon many times in a row.

I'm a bit unsure whether or not we should change that, or leave it up to the implementation to use the schema type icon 🤔

Reply:
OK. Where would it be repeated?
Should I get the icon like in the code in my previous comment?

Reply:
In the global "create new document" modal, I think we made a choice not to repeat the icon. 

Yeah, I think for now you can just use the code you outlined, and we can consider changing/cleaning it up when we have more time.